### PR TITLE
[XLA:CPU] Layer Norm XLA HLO Pattern Matcher with oneDNN custom call rewrite

### PR DESCRIPTION
**Pull request by akhilgoe**
This PR enables OneDNN library call for the matched XLA HLO Layer Norm pattern through custom_call instruction. In particular, this PR adds:

1. oneDNN ops rewriter pass that will rewrite custom oneDNN ops like Layer Norm and Softmax. Currently, this pass is located before the BF16 all-reduce to F32 promotion.
2. Layer Norm pattern matcher as seen in some Flax models
3. oneDNN Layer Norm custom call rewrite
4. Variable number of arguments for oneDNN custom call targets (by @mdfaijul )
5. A test for XLA HLO Layer Norm pattern match and rewrite verification

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @akhilgoe Can you please rebase your branch and resolve conflicts? Thank you!

**Comment by akhilgoel999** [comment]
Hi @gbaned...the conflicts have been resolved. Thanks!

File: tensorflow/compiler/xla/tests/onednn_layer_norm_test.cc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    multiply.9770 = f32[4,1,256]{2,1,0} multiply(broadcast.9766, broadcast.9769)
+    multiply.9771 = f32[4,1,256]{2,1,0} multiply(subtract.9761, multiply.9770)
+    broadcast.9772 = f32[1,1,256]{2,1,0} broadcast(Arg_0.3), dimensions={0,1,2}
+    reshape.9773 = f32[1,256]{1,0} reshape(broadcast.9772)
+    broadcast.9774 = f32[4,1,256]{2,1,0} broadcast(reshape.9773), dimensions={1,2}
+    ROOT add.9775 = f32[4,1,256]{2,1,0} add(multiply.9771, broadcast.9774)
+  }  
+)";
+
+  EXPECT_TRUE(RunAndCompare(layer_norm_module_str, ErrorSpec{1e-4, 1e-4}));
```
Comment (d0k) [inline tensorflow/compiler/xla/tests/onednn_layer_norm_test.cc:87]: Can this test somehow that onednn actually gets executed? I'm worried that we'll never notice when the onednn rewriter breaks.

**Comment by gbaned** [comment]
Hi @akhilgoe Can you please check @d0k's comments and keep us posted ? Thank you!

File: tensorflow/compiler/xla/tests/onednn_layer_norm_test.cc (hunk 1 above)
Comment (akhilgoel999) [reply tensorflow/compiler/xla/tests/onednn_layer_norm_test.cc:87]: @mdfaijul @agramesh1 please help. Thanks!

Comment (mahmoud-abuzaina) [reply tensorflow/compiler/xla/tests/onednn_layer_norm_test.cc]: Thanks @d0k. I have updated the tests to check the rewrite as suggested.

**Review by d0k — APPROVED** [review]

**Comment by mahmoud-abuzaina** [comment]
Hi @penpornk, could you help to review/merge this PR?

**Review by gbaned — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @akhilgoe Can you please rebase your branch and resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @akhilgoe Can you please rebase your branch and resolve conflicts? Thank you!

**Comment by mahmoud-abuzaina** [comment]
I have resolved the merge conflicts.

**Review by d0k — APPROVED** [review]

**Comment by gbaned** [comment]
Hi @akhilgoe Can you please rebase your branch and resolve the conflicts? Thank you!

**Comment by mahmoud-abuzaina** [comment]
Replaced by https://github.com/openxla/xla/pull/7485. please close this PR.

**Comment by penpornk** [comment]
Closing this PR as its replacement (https://github.com/openxla/xla/pull/7485) has been merged.
